### ISSUE TITLE
"disablePathToLower" does not work as expected in 0.21

### ISSUE THREAD

**Issue by xofyarg** [spf13/hugo#3374]
```
# commit 793554108763c0984f1a1b1a6ee5744b560d78d0

$ grep -i tolower config.yaml
disablePathToLower: true

$ find content/Test/
content/Test/
content/Test/Post
content/Test/Post/index.md

$ find public/Test/ public/test/
public/Test/
public/Test/index.html
public/Test/index.xml
public/test
public/test/post
public/test/post/index
public/test/post/index/index.html
```

**Comment by f00b4r0** [comment]
I can confirm this problem in 0.21

**Comment by f00b4r0** [comment]
Dug a little more, this only affects the directory part of the path. When using uglyURLs, if disablePathToLower is set to true, `content/Test/Toto.md` generates `public/test/Toto.html`; and `public/test/toto.html` if disablePathToLower isn't set.

**I know nothing about Go**, but I suspect the problem could be coming from this [code line](https://github.com/spf13/hugo/blob/e28d9aa42c3429d22fe254e69e4605aaf1e684f3/hugolib/page_paths.go#L88)

HTH

**Comment by f00b4r0** [comment]
The following hack fixes the issue for me, so this line is effectively the source of the issue.

```
diff --git a/hugolib/page_paths.go b/hugolib/page_paths.go
index 1fe8ea5..0f5add2 100644
--- a/hugolib/page_paths.go
+++ b/hugolib/page_paths.go
@@ -85,7 +85,7 @@ func (p *Page) initTargetPathDescriptor() error {
                Kind:     p.Kind,
                Sections: p.sections,
                UglyURLs: p.s.Info.uglyURLs,
-               Dir:      filepath.ToSlash(strings.ToLower(p.Source.Dir())),
+               Dir:      filepath.ToSlash(p.Source.Dir()),
                URL:      p.URLPath.URL,
        }
```

 I'm unable to tell whether this is a sufficient fix or if something else was meant to happen here.

HTH

**Comment by bep** [comment]
>I'm unable to tell whether this is a sufficient fix or if something else was meant to happen here.

As I'm assuming you have tested only your concern, I would assume not.

**Comment by f00b4r0** [comment]
As I said I know nothing about go. All I can provide is a clue where the problem is (took me a quick git grep to find that out) and confirm that the problem is really coming from here, which I did.

I tested both states of `disablePathToLower` and both states of `uglyURLs`, and it seems to work as intended now.

I'm merely trying to make progress with this issue which affects the website I wanted to convert to hugo...

Hope that helps still.
